WordPress/Helpers/SanitizationFunctionsTrait.php

```diff
@@ -0,0 +1,238 @@
+		'sanitize_sql_orderby'       => true,
+		'sanitize_term_field'        => true,
+		'sanitize_term'              => true,
+		'sanitize_text_field'        => true,
+		'sanitize_textarea_field'    => true,
+		'sanitize_title_for_query'   => true,
+		'sanitize_title_with_dashes' => true,
+		'sanitize_title'             => true,
+		'sanitize_user_field'        => true,
+		'sanitize_user'              => true,
+		'validate_file'              => true,
+		'wp_handle_sideload'         => true,
+		'wp_handle_upload'           => true,
+		'wp_kses_allowed_html'       => true,
+		'wp_kses_data'               => true,
+		'wp_kses_post'               => true,
+		'wp_kses'                    => true,
+		'wp_parse_id_list'           => true,
+		'wp_redirect'                => true,
+		'wp_safe_redirect'           => true,
+		'wp_sanitize_redirect'       => true,
+		'wp_strip_all_tags'          => true,
+	);
+
+	/**
+	 * Sanitizing functions that implicitly unslash the data passed to them.
+	 *
+	 * This list is complementary to the `$sanitizingFunctions` list.
+	 * Sanitizing functions should be added to this list if they also
+	 * implicitely unslash data and to the `$sanitizingFunctions` list
+	 * if they don't.
+	 *
+	 * @since 0.5.0
+	 * @since 0.11.0 Changed from public static to protected non-static.
+	 * @since 3.0.0  - Moved from the Sniff class to this trait.
+	 *               - Visibility changed from protected to private.
+	 *
+	 * @var array<string, bool>
+	 */
+	private $unslashingSanitizingFunctions = array(
+		'absint'       => true,
+		'boolval'      => true,
+		'count'        => true,
+		'doubleval'    => true,
+		'floatval'     => true,
+		'intval'       => true,
+		'sanitize_key' => true,
+		'sizeof'       => true,
+	);
+
+	/**
+	 * Cache of previously added custom functions.
+	 *
+	 * Prevents having to do the same merges over and over again.
+	 *
+	 * @since 0.4.0
+	 * @since 0.11.0 - Changed from public static to protected non-static.
+	 *               - Changed the format from simple bool to array.
+	 * @since 3.0.0  - Moved from the NonceVerification and the ValidatedSanitizedInput sniff classes to this class.
+	 *               - Visibility changed from protected to private.
+	 *
+	 * @var array<string, string[]>
+	 */
+	private $addedCustomSanitizingFunctions = array(
+		'sanitize'        => array(),
+		'unslashsanitize' => array(),
+	);
+
+	/**
+	 * Combined list of WP/PHP native and custom sanitization functions.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var array<string, bool>
+	 */
+	private $allSanitizingFunctions = array();
+
+	/**
+	 * Combined list of WP/PHP native and custom sanitization and unslashing functions.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var array<string, bool>
+	 */
+	private $allUnslashingSanitizingFunctions = array();
+
+	/**
+	 * Retrieve a list of all known sanitization functions.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var array<string, bool>
+	 */
+	public function get_sanitization_functions() {
```

Comment:
How come this is function named `sanitization` and not `sanitizING`? Seems inconsistent with the property names as well as "escapING", "array walkING", and other -ING references.

Reply:
Grammar technically, "sanitization" sounds better to me that "sanitizing", but happy enough to change it (in all the places in the trait).

Reply:
If we change this, should the trait name also be changed to `SanitizingFunctionsTrait` ?

Reply:
I've adjusted this in a separate commit now. Let me know what you think and i can squash the commits if okay.

Reply:
Looks good - good to squash.

Reply:
Squashed + updated the commit message and the above PR description to use the new Trait and method names.